accepted/2021/install-location-per-architecture.md

```diff
@@ -138,34 +141,56 @@ any specific knowledge of where to find the install location, it would just
+on Linux and macOS.
 
-*Note: If installers write the file, they should also read it and validate
-that they don't overwrite it with different information. It's unclear yet
-if this is necessary and in which situations.*
+#### Windows installer behavior
 
-#### Support for downlevel apphost
+Pre-.NET 6 Windows installer writes registry keys for all architectures
+(even though it only installs one of them). Ideally the installer would
+only write the key for the architecture it's actually installing.
 
-In order to support running downlevel apphost apps on multi-arch non-Windows
-architectures (currently only applies to macOS), the installer of .NET 6 and above
-should write the configuration file in a specific shape, for example (on macOS arm64):
+#### Linux/macOS installer behavior
 
-```console
-/usr/local/share/dotnet/x64
-arm64=/usr/local/share/dotnet
-x64=/usr/local/share/dotnet/x64
-```
+Pre-.NET 6 Linux/macOS installer doesn't write the `install_location` file.
 
-With the above proposed changes this would guarantee that all apphosts
-would work if the right frameworks are installed.
+Starting with .NET 6, installers should write architecture specific file
+with the product location, in the format `install_location_<arch>`. So for example
+`install_location_x64` (all lower case). The content of the file should be
+a single line with the path to the product.
 
-* 3.1 and 5 apphost would only read the first line, which points to x64 install.
-Since 3.1 and 5 are only supported on x64 on macOS, that is the right location.
-* arm64 6 apphost would react to the `arm64` location
-* x64 6 apphost would react to the `x64` location
+*Note: If installers can write the file, they should also read it and validate
+that they don't overwrite it with different information.*
+
+#### Support for downlevel apphost
+
+In order to support running downlevel apphost apps on multi-arch non-Windows
+architectures (currently only applies to macOS), the installer for
+the architecture which .NET supports in versions lower than .NET 6
+should also write the product location to the arch-less location file
+`install_location`. This is needed for downlevel applications to correctly
+find the product.
+
+For example this should be present on macOS arm64 systems if both x64 and arm64
+products are installed.
+
+| File name | Content |
+| --- | --- |
+| `install_location` | `usr/local/share/dotnet/x64` |
+| `install_location_arm64` | `usr/local/share/dotnet` |
+| `install_location_x64` | `usr/local/share/dotnet/x64` |
+
+* 3.1 and 5 apphosts will only read the arch-less file `install_location`,
+which points to x64 install. Since 3.1 and 5 are only supported on x64 on macOS,
+that is the right location.
+* arm64 6 apphost will react to the `install_location_arm64`
+* x64 6 apphost will react to the `install_location_x64`
+
+*Note: All supported downlevel installers (3.1 and 5) will need to be updated
```

Comment:
.NET 5 will not get an update per https://github.com/dotnet/designs/blob/main/accepted/2021/x64-emulation-on-arm64/x64-emulation.md.  Not sure if we're making the same concession to the host or not. @richlander 
```suggestion
*Note: All supported downlevel installers (3.1) will need to be updated
```

Reply:
We're not updating any hosts with this (it would be sort of moot, we can't really ask people to rebuild all of their osx-x64 apps to make them work on macOS arm64, so it has to work on its own without changes).